pkg/tests/ossm/ior_test.go

```diff
@@ -0,0 +1,322 @@
+package ossm
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
+)
+
+type RouteMetadata struct {
+	Uid             string            `json:"uid"`
+	Name            string            `json:"name"`
+	ResourceVersion string            `json:"resourceVersion"`
+	Labels          map[string]string `json:"labels"`
+	Annotations     map[string]string `json:"annotations"`
+}
+
+type Route struct {
+	Metadata RouteMetadata `json:"metadata"`
+}
+
+const (
+	originalHostAnnotation = "maistra.io/original-host"
+)
+
+// TestIOR tests IOR error regarding routes recreated: https://issues.redhat.com/browse/OSSM-1974. IOR will be deprecated on 2.4 and willl be removed on 3.0
+func TestIOR(t *testing.T) {
+	test.NewTest(t).Groups(test.Full).Run(func(t test.TestHelper) {
+		t.Log("This test verifies the behavior of IOR.")
+
+		meshNamespace := env.GetDefaultMeshNamespace()
+		meshName := env.GetDefaultSMCPName()
+
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, meshNamespace)
+		})
+
+		host := "www.test.ocp"
+
+		createSimpleGateway := func(t test.TestHelper) {
+			t.Logf("Creating Gateway for %s host", host)
+			oc.ApplyString(t, "", generateGateway("gw", meshNamespace, host))
+		}
+
+		checkSimpleGateway := func(t test.TestHelper) {
+			t.Logf("Checking whether a Route is generated for %s", host)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				routes := getRoutes(t, meshNamespace)
+				found := routes[0].Metadata.Annotations[originalHostAnnotation]
+				if found != host {
+					t.Fatalf("Expect a route set for %s host, but got %s instead", host, found)
+				} else {
+					t.LogSuccessf("Got an expected Route for %s", host)
+				}
+			})
+		}
+
+		DeployControlPlane(t)
+
+		t.NewSubTest("check IOR off by default v2.4").Run(func(t test.TestHelper) {
```

Comment:
IOR will remain ON by default until v2.5.

Reply:
Delete the subtest case because is not needed until 2.5